cmake/modules/BuildBoost.cmake

```diff
@@ -303,5 +303,8 @@ endfunction()
 
 function(add_executable target)
   _add_executable(${target} ${ARGN})
-  maybe_add_boost_dep(${target})
+  # can't add dependencies to aliases
+  if (NOT ";${ARGN};" MATCHES ";(ALIAS);")
```

Comment:
@clwluvw hi Seena, do we have any ALIAS target?

Reply:
hi @tchaikov, I faced it when I was developing on project, and then the alias was failing as it was trying to add boost dep. The same thing is done for the `add_library` override. I just thought it might make sense to have this here as well (although currently there are no aliases failing because of this) but just logically... :)
Feel free to close it - was not a big deal at all.

Reply:
i see. so this change was created to address the correctness issue which might raise in future, but not a problem in existing source tree, so that 
```cmake
add_executable(<name> ALIAS <target>)
```
works as expected.

if you are willing to continue working on this pull request. i'd suggest put the rationales in the commit message, like what's the impact of this fix. it currently reads

> Can't add dependencies to aliases

it helps. but it does not address the doubt that i had and probably the posterity might have in future.

Reply:
thanks, @tchaikov - I have edited my commit message. Can you please take another look?